components/core/src/os/process/windows.rs

```diff
@@ -234,15 +219,85 @@ impl Child {
             }
         }
 
-        // turn Ctrl-C handling back on for current process
-        ret = unsafe { kernel32::SetConsoleCtrlHandler(None, winapi::FALSE) };
-        if ret == 0 {
-            debug!("Failed to call SetConsoleCtrlHandler on pid {}: {}",
-                   self.pid,
-                   io::Error::last_os_error());
+        result
+    }
+
+    fn terminate_process_descendants(&self,
+                                     table: &HashMap<winapi::DWORD, Vec<winapi::DWORD>>,
+                                     pid: winapi::DWORD)
+                                     -> Result<()> {
+        if let Some(children) = table.get(&pid) {
+            for child in children {
+                self.terminate_process_descendants(table, child.clone())?;
+            }
+        }
+        unsafe {
+            match handle_from_pid(pid) {
+                Some(h) => {
+                    let ret = kernel32::TerminateProcess(h, 1);
+                    if ret == 0 {
+                        return Err(Error::TerminateProcessFailed(format!("Failed to call \
+                                                                        terminate pid {}: {}",
```

Comment:
Really minor, but you might want to put the API name 'TerminateProcess' in the error message (like you do for the CreateToolhelp32Snapshot message err below).